pkg/controllers/dynakube/istio/reconciler.go

```diff
@@ -157,40 +221,62 @@ func (r *reconciler) reconcileIPServiceEntry(ctx context.Context, ipHosts []dtcl
 	return nil
 }
 
+func (r *reconciler) cleanupIPServiceEntry(ctx context.Context, component string) error {
+	entryName := BuildNameForIPServiceEntry(r.client.Owner.GetName(), component)
+
+	err := r.client.DeleteServiceEntry(ctx, entryName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
 func (r *reconciler) reconcileFQDNServiceEntry(ctx context.Context, fqdnHosts []dtclient.CommunicationHost, component string) error {
 	owner := r.client.Owner
 	entryName := BuildNameForFQDNServiceEntry(owner.GetName(), component)
 
 	if len(fqdnHosts) != 0 {
-		meta := buildObjectMeta(
+		objectMeta := buildObjectMeta(
 			entryName,
 			owner.GetNamespace(),
 			labels.NewCoreLabels(owner.GetName(), component).BuildLabels(),
 		)
 
-		serviceEntry := buildServiceEntryFQDNs(meta, fqdnHosts)
+		serviceEntry := buildServiceEntryFQDNs(objectMeta, fqdnHosts)
 
 		err := r.client.CreateOrUpdateServiceEntry(ctx, serviceEntry)
 		if err != nil {
 			return err
 		}
 
-		virtualService := buildVirtualService(meta, fqdnHosts)
+		virtualService := buildVirtualService(objectMeta, fqdnHosts)
 
 		err = r.client.CreateOrUpdateVirtualService(ctx, virtualService)
 		if err != nil {
 			return err
 		}
 	} else {
-		err := r.client.DeleteServiceEntry(ctx, entryName)
+		err := r.cleanupFQDNServiceEntry(ctx, component)
 		if err != nil {
 			return err
 		}
+	}
 
-		err = r.client.DeleteVirtualService(ctx, entryName)
-		if err != nil {
-			return err
-		}
+	return nil
+}
+
+func (r *reconciler) cleanupFQDNServiceEntry(ctx context.Context, component string) error {
+	entryName := BuildNameForFQDNServiceEntry(r.client.Owner.GetName(), component)
+
+	err := r.client.DeleteServiceEntry(ctx, entryName)
+	if err != nil {
+		return err
+	}
+
+	err = r.client.DeleteVirtualService(ctx, entryName)
+	if err != nil {
+		return err
 	}
```

Comment:
here if you was not able to delete SE, we never try to delete VS. May be you should use same trick you did with errors.Join?
```
package main

import (
	"errors"
	"fmt"
)

func main() {
	err1 := errors.New("err1")
	var err2 error
	err := errors.Join(err1, err2)
	fmt.Println(err)
	if errors.Is(err, err1) {
		fmt.Println("err is err1")
	}
}
```

Reply:
I like it, should improve cleanup.

Reply:
fixed in c09681fc